src/postgraphile/shutdownActions.ts

```diff
@@ -0,0 +1,84 @@
+export type Action = () => void | Promise<void>;
+
+/**
+ * This class tracks actions that must be taken when PostGraphile is shut down
+ * (released) in order to make sure that all the resources it has consumed has
+ * been cleaned up.
+ *
+ * Since PostGraphile is extensible via plugins and plugins may have their own
+ * shutdown actions we use this generic system to handle them all. If your
+ * plugin sets up something that would keep running after PostGraphile is
+ * `.release()`d (such as a timer, interval, network connection, filesystem
+ * monitoring, or similar activity) then you should register an action via
+ * `shutdownActions` to clean it up.
```

Comment:
```suggestion
 * `shutdownActions` to clean it up.
 * 
 * This class currently has "experimental" status, it may have breaking
 * changes in future semver minor releases.
```